src/java/org/apache/cassandra/metrics/MessagingMetrics.java

```diff
@@ -98,6 +98,9 @@ private static final class DroppedForVerb
 
     // total dropped message counts for server lifetime
     private final Map<Verb, DroppedForVerb> droppedMessages = new ConcurrentHashMap<>();
+    
+    // dropped mutations by table
+    private final Map<String, DroppedMessageMetrics> droppedMutationsByTable = new ConcurrentHashMap<>();
```

Comment:
compared to a copy-on-write approach, what's the performance implications here ? 

while this only hits when mutations are dropped, so off the hot path for writes as far as the user cares, might this add unnecessary load in a ~cascading failure situation (for an otherwise static schema) ? 

imagine 10k/s mutations all dropped in the same 10 second period, that's a lot of calls on `computeIfAbsent`